From 58d24e21dffee3e82060ee38981a9c5397e34ff3 Mon Sep 17 00:00:00 2001
From: Niall Kegan Pemberton <niallp@apache.org>
Date: Thu, 23 Nov 2006 13:24:38 +0000
Subject: [PATCH] VALIDATOR-208 - JavaScript function jcv_isFieldPresent()
 causes error in IE5 using "undefined" - thanks to Leo Asanov

git-svn-id: https://svn.apache.org/repos/asf/jakarta/commons/proper/validator/trunk@478566 13f79535-47bb-0310-9956-ffa450edef68
---
 .../apache/commons/validator/javascript/validateUtilities.js   | 2 +-
 xdocs/changes.xml                                              | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/javascript/org/apache/commons/validator/javascript/validateUtilities.js b/src/javascript/org/apache/commons/validator/javascript/validateUtilities.js
index 63df1dc24..fe2757c57 100644
--- a/src/javascript/org/apache/commons/validator/javascript/validateUtilities.js
+++ b/src/javascript/org/apache/commons/validator/javascript/validateUtilities.js
@@ -87,7 +87,7 @@
    */
   function jcv_isFieldPresent(field) {
       var fieldPresent = true;
-      if (field == null || field == undefined) {
+      if (field == null || (typeof field == 'undefined')) {
           fieldPresent = false;
       } else {
           if (field.disabled) {
diff --git a/xdocs/changes.xml b/xdocs/changes.xml
index fa3d652f9..229b9712e 100644
--- a/xdocs/changes.xml
+++ b/xdocs/changes.xml
@@ -40,6 +40,9 @@ The <action> type attribute can be add,update,fix,remove.
   <body>
 
     <release version="1.3.1" date="Pending">
+      <action dev="niallp" type="fix" issue="VALIDATOR-208" due-to="Leo Asanov">
+        JavaScript function jcv_isFieldPresent() causes error in IE5 using "undefined".
+      </action>
       <action dev="niallp" type="fix" issue="VALIDATOR-190">
         EmailValidator allows control characters (ASCII 0-31 and 127).
         <dueto name="Cott"/>
